src/test/java/org/jenkinsci/plugins/docker/workflow/client/DockerClientTest.java

```diff
@@ -76,6 +76,26 @@ public void test_run() throws IOException, InterruptedException {
     }
 
     @Test
+    public void test_run_without_user() throws IOException, InterruptedException {
+        EnvVars launchEnv = newLaunchEnv();
+        String containerId =
+                dockerClient.run(launchEnv, "learn/tutorial", null, null, Collections.<String, String>emptyMap(), Collections.<String>emptyList(), new EnvVars(),
+                        null, "echo", "hello world");
+        Assert.assertEquals(64, containerId.length());
+        ContainerRecord containerRecord = dockerClient.getContainerRecord(launchEnv, containerId);
+        Assert.assertEquals(dockerClient.inspect(launchEnv, "learn/tutorial", ".Id"), containerRecord.getImageId());
+        Assert.assertTrue(containerRecord.getContainerName().length() > 0);
+        Assert.assertTrue(containerRecord.getHost().length() > 0);
+        Assert.assertTrue(containerRecord.getCreated() > 1000000000000L);
+        Assert.assertEquals(Collections.<String>emptyList(), dockerClient.getVolumes(launchEnv, containerId));
```

Comment:
Not really testing anything interesting here; maybe better to just delete this, especially if the `user` argument is deleted.

What I would like to see in a test (if it is possible) is verification in `WithContainerStepTest` that you are fixing some bug: a script which fails before the switch of the user ID from `run` to `exec` but which passes afterwards.